kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/NumberFunction.java

```diff
@@ -25,23 +32,30 @@ public NumberFunction() {
         super( "number" );
     }
 
-    public BigDecimal apply(@ParameterName("from") String from, @ParameterName("grouping separator") String group, @ParameterName("decimal separator") String decimal) {
+    public Either<FEELEvent, BigDecimal> apply(@ParameterName("from") String from, @ParameterName("grouping separator") String group, @ParameterName("decimal separator") String decimal) {
         if ( from == null ) {
-            return null;
+            return Either.ofLeft(new InvalidParametersEvent(Severity.ERROR, "from", "cannot be null"));
         }
         if ( group != null && !group.equals( " " ) && !group.equals( "." ) && !group.equals( "," ) ) {
-            return null;
+            return Either.ofLeft(new InvalidParametersEvent(Severity.ERROR, "group", "not a valid one, can only be one of: dot ('.'), comma (','), space (' ') "));
         }
         if ( decimal != null && ((!decimal.equals( "." ) && !decimal.equals( "," )) || (group != null && decimal.equals( group ))) ) {
-            return null;
+            return Either.ofLeft(new InvalidParametersEvent(Severity.ERROR, "decimal", "invalid parameter 'decimal' used in conjuction with specified parameter 'group'"));
         }
+        
         if ( group != null ) {
             from = from.replaceAll( "\\" + group, "" );
         }
         if ( decimal != null ) {
             from = from.replaceAll( "\\" + decimal, "." );
         }
-        return new BigDecimal( from );
+        
+        try {
+            // TODO: here cannot call EvalHelper.bigDecimalOrNull as it doesn't support String as parameter for input, but then how to set MathContext.DECIMAL128 ?
```

Comment:
@etirelli can we discuss this as per the comment I've noted, please? As agreed I've checked if there was some stray BigDecimal initialization and this is one I could not correct it.